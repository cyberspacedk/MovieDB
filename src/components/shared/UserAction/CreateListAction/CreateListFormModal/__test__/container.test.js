import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import CreateListFormModalConnected from '../container';

describe('CreateListActionConnected ', () => {
  const store = configureStore()({
    myLists: {
      ids: [1, 2, 3],
    },
    database: {
      lists: { 1: {} },
    },
  });
  store.dispatch = jest.fn();

  const props = {
    hideModal: jest.fn(),
    handleReset: jest.fn(),
    handleSubmit: jest.fn(),
    isSubmitting: jest.fn(),
    status: 'waiting',
  };

  const container = shallow(
    <CreateListFormModalConnected store={store} {...props} />,
  );

  it('Should match its snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  // необходимо протестировать остальные функции
});
