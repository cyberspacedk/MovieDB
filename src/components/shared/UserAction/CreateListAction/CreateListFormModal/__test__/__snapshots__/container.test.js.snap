// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateListActionConnected  Test container with connect Should match its snapshot 1`] = `
<CreateListFormModal
  createListRequest={[Function]}
  dirty={false}
  errors={Object {}}
  handleBlur={[Function]}
  handleChange={[Function]}
  handleFormCancel={[Function]}
  handleFormSubmit={[Function]}
  handleReset={[Function]}
  handleSubmit={[Function]}
  initialValues={
    Object {
      "description": "",
      "name": "",
    }
  }
  isSubmitting={false}
  isValid={false}
  isValidating={false}
  registerField={[Function]}
  resetForm={[Function]}
  setError={[Function]}
  setErrors={[Function]}
  setFieldError={[Function]}
  setFieldTouched={[Function]}
  setFieldValue={[Function]}
  setFormikState={[Function]}
  setStatus={[Function]}
  setSubmitting={[Function]}
  setTouched={[Function]}
  setValues={[Function]}
  status="WAITING"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  submitCount={0}
  submitForm={[Function]}
  touched={Object {}}
  unregisterField={[Function]}
  validateField={[Function]}
  validateForm={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
  values={
    Object {
      "description": "",
      "name": "",
    }
  }
/>
`;

exports[`CreateListActionConnected  Test container with connect check validationSchema 1`] = `
ObjectSchema {
  "_blacklist": RefSet {
    "list": Set {},
    "refs": Map {},
  },
  "_conditions": Array [],
  "_defaultDefault": [Function],
  "_deps": Array [],
  "_excludedEdges": Array [],
  "_exclusive": Object {},
  "_mutate": undefined,
  "_nodes": Array [
    "description",
    "name",
  ],
  "_options": Object {
    "abortEarly": true,
    "recursive": true,
  },
  "_type": "object",
  "_typeError": [Function],
  "_whitelist": RefSet {
    "list": Set {},
    "refs": Map {},
  },
  "fields": Object {
    "description": StringSchema {
      "_blacklist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "_conditions": Array [],
      "_deps": Array [],
      "_exclusive": Object {
        "min": true,
        "required": true,
      },
      "_mutate": undefined,
      "_options": Object {
        "abortEarly": true,
        "recursive": true,
      },
      "_type": "string",
      "_typeError": [Function],
      "_whitelist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "tests": Array [
        [Function],
        [Function],
      ],
      "transforms": Array [
        [Function],
      ],
    },
    "name": StringSchema {
      "_blacklist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "_conditions": Array [],
      "_deps": Array [],
      "_exclusive": Object {
        "min": true,
        "required": true,
        "undefined": false,
      },
      "_mutate": undefined,
      "_options": Object {
        "abortEarly": true,
        "recursive": true,
      },
      "_type": "string",
      "_typeError": [Function],
      "_whitelist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "tests": Array [
        [Function],
        [Function],
        [Function],
      ],
      "transforms": Array [
        [Function],
      ],
    },
  },
  "tests": Array [],
  "transforms": Array [
    [Function],
  ],
}
`;
